### ISSUE TITLE
Name property

### ISSUE THREAD

**Issue by simon-p-r** [nlf/wadofgum#3]
Hi @nlf 

Can the base class have its own name property attached to meta which needs to be passed to constructor to create.  Currently "this.constructor.name" is relied upon, see here https://github.com/nlf/wadofgum-nedb/blob/master/lib/index.js#L54 which won't be set if class is extended anonymously.  I create wadofgum models dynamically and the class key word doesn't accept computed property names meaning I have to do something like this.  

```js
let [computedProp] = class extends Wadofgum.mixin(Validation) {};
````

which means "this.constructor.name" is undefined

I can do a PR if agreeable? 

Oh yeah thanks for this awesome module!  It is genius!

**Comment by nlf** [comment]
yeah, i think allowing for an optionally passed in name would be fine. i'd like to maintain the current behavior if no name is passed in, but if one is store and use it instead. go ahead and take a try at a PR and let's see how it looks.

i'm glad you like the module! it's been kind of a toy project for some time now, but i think it's really shaping into something useful. definitely keep opening issues for any concerns or suggestions!

**Comment by nlf** [comment]
hmm.. it occurs to me that passing the name in the constructor won't necessarily work as expected. a lot of code expects the class itself to have a name property as well, which won't exist if it's assigned in the constructor.

instead we may have to use a getter and setter on the class, where the getter returns something like `this.name || this.meta.name` and it'll have to be used like

```js
let model = class extends Wadofgum {};
model.name = 'Something';
```

we can't use the `name` property though since that's a reserved property on a function (which is all a class really is) so we'll have to think of something else

**Comment by simon-p-r** [comment]
Examples I have seen use **_name** as an alternative but that maybe confusing.

**Comment by nlf** [comment]
yeah, i'm not a fan of prefixing things with an underscore unless they're not intended to be used by the end user..

maybe `type`?

**Comment by simon-p-r** [comment]
Ok that was what I was thinking

**Comment by nlf** [comment]
closed via #4
